src/main/java/com/google/sps/servlets/ReviewsServlet.java

```diff
@@ -73,6 +78,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
     }
   }
 
+  /**
+   * Recalculates the mean (aggregate) rating for the current Business entity, based on the new review.
+   * @param businessEntity business entity to update
+   * @param numReviews total number of existing reviews, including the one just added
+   * @param newRating the rating provided in the newest review
+   * @return a float for the new aggregate rating
+   */
+  private float recalculateRating(Entity businessEntity, int numReviews, int newRating) {
+    float oldAggregate = Float.parseFloat((String) businessEntity.getProperty(BUSINESS_RATING));
+    return (oldAggregate*(numReviews-1) + newRating) / numReviews;
```

Comment:
I think we can add a null check on this for the 0 review case!

Reply:
Checked for NOT_FOUND as Dalis suggested!